src/main/scala/vaquita/VecALU.scala

```diff
@@ -0,0 +1,462 @@
+    val mask_bit_undisturb = mask_vs0===0.B && io.mask_arith===0.B && vsetvli_mask===0.B
+    val vec_sew8_result = WireInit(0.S(8.W))
+    dontTouch(vec_sew8_result)
+    when(io.alu_opcode==="b010000".U || io.alu_opcode==="b010010".U){
+      vec_sew8_result := (Arithmatic(vs1.asSInt, vs2.asSInt,vs3,8,mask_vs0.asUInt)).asSInt
+      }.otherwise{
+      vec_sew8_result := Mux(mask_bit_active_element===1.B,Arithmatic(vs1.asSInt, vs2.asSInt,vs3,8,mask_vs0.asUInt),Mux(mask_bit_undisturb===1.B,vs3,Fill(16,1.U).asSInt)).asSInt
+      }
+    vec_sew8_result 
+  }
+
+  // Function for vslideup
+  def vslideup(input: SInt, slide_amount:UInt,mask_vs0:Bool,vs3:SInt,i_incre:UInt,sew:Int,count_sew:Int): SInt = {
+    val shifted                 = Wire(SInt(32.W))
+    val in_i                    = WireInit(0.U(33.W))
+    val in_j                    = WireInit(0.U(10.W))
+    val sew1                    = WireInit(0.U(10.W))
+    val count_sew1              = WireInit(0.U(10.W))
+    val vs2_element             = WireInit(0.S(sew.W))
+    val sewup                   = WireInit(0.U(10.W))
+    val sewdown                 = WireInit(0.U(10.W))
+    val vsetvli_mask            = 0.B
+    val mask_bit_active_element = (mask_vs0===1.B && io.mask_arith===0.B) || io.mask_arith===1.B
+    val mask_bit_undisturb      = mask_vs0===0.B && io.mask_arith===0.B && vsetvli_mask===0.B
+    sewup       := ((count_sew*sew)-1).U
+    sewdown     := ((count_sew-1)*sew).U
+    sew1        := sew.U
+    count_sew1  := count_sew.U
+  // Mux(mask_bit_active_element===1.B,Arithmatic(vs1, vs2,vs3,32,mask_vs0.asUInt),Mux(mask_bit_undisturb===1.B,vs3,Fill(32,1.U).asSInt)).asSInt
+  when(count_sew1===1.U && io.sew==="b000".U){
+    in_i := ((slide_amount)/(config.vlen.U/32.U)+i_incre)
+    in_j := (slide_amount)%(config.vlen.U/32.U)
+  }.elsewhen(count_sew1===2.U && io.sew==="b000".U){
+    in_i := (slide_amount)/(config.vlen.U/32.U)+i_incre
+    in_j := (slide_amount)%(config.vlen.U/32.U)
+  }
+  .otherwise{
+    in_i := (slide_amount)/(config.vlen.U/32.U)+i_incre
+    in_j := (slide_amount)%(config.vlen.U/32.U)
+  }
+    vs2_element := (io.vs2_in(in_i)(in_j)((count_sew*sew)-1,(count_sew-1)*sew).asSInt)
+    shifted := Mux(mask_bit_active_element===1.B,vs2_element,Mux(mask_bit_undisturb===1.B,vs3,Fill(sew,1.U).asSInt)).asSInt
+    shifted
+  }
+
+
+
+// Function for vslidedown
+  // def vslidedown(input: SInt, slide_amount: UInt): SInt = {
+  //   val shifted = WireInit(0.S(32.W))
+  // // shifted := Mux(mask===1.B && io.mask_arith===0.B || io.mask_arith===1.B,(io.vs2_in(in_i)(in_j)).asSInt, vs3)
+  //   shifted
+  // }
+
+// // Function for vrgather
+//   def vrgather(input: Vec[Vec[SInt]], indices: Vec[Vec[UInt]], i: Int, j: Int): SInt = {
+//   // Gathers element from input based on indices provided in indices vector
+//   // Ensures index is within bounds, else returns zero
+//     val index = indices(i)(j)
+//     Mux(index < input.length.U, input(i)(index), 0.S)
+//   }
```

Comment:
too much commented code i am seeing, if you think it is going to be needed in future, take it to the end of the file or jus delete it